### PR TITLE
Feat/multiclass segmentation

### LINKED ISSUE #48: Migrate Multiclass Semantic Segmentation
## Description
Currently, the URC repository does not utilize machine learning of any kind for obstacle detection. A good way to kickstart our ML efforts would be to migrate the vision based perception strategy from the IGVC repository.

The vision based perception strategy in the IGVC repository relies on a [UNET](https://arxiv.org/abs/1505.04597) with an [EfficientNet](https://arxiv.org/abs/1905.11946) encoder to perform semantic segmentation on input images. The neural net for the IGVC competition was trained to classify pixels as line, barrel, or ground. For the URC, the network will probably have to be retrained to classify obstacles for the URC competition. Retraining will involve some effort to label images.

Our encoder is pretrained on the [ImageNet](https://en.m.wikipedia.org/wiki/ImageNet) dataset which allows us to utilize transfer learning to train our model. Networks trained from scratch need a massive amounts of data and compute. Even with the appropriate dataset and compute you would need to wait days or weeks to train such a model. Using this strategy we can get away with using 500ish images (versus millions of images needed if training from scratch) to fine tune the weights of the pretrained encoder and achieve highly accurate results.

## Requirements

 - Migrate [multiclass_segmentation.py](https://github.com/RoboJackets/igvc-software/blob/c4de30fc741c9fb3d0923596bb2263d947eec3dd/igvc_perception/src/multiclass_segmentation/multiclass_segmentation.py) and the corresponding [multiclass segmentation folder](https://github.com/RoboJackets/igvc-software/tree/master/igvc_perception/src/multiclass_segmentation) 
 - Migrate [multiclass_segmentation.launch](https://github.com/RoboJackets/igvc-software/blob/c4de30fc741c9fb3d0923596bb2263d947eec3dd/igvc_perception/launch/multiclass_segmentation.launch)
 - Successfully launches

### PR DESCRIPTION
**Pull request by a-stickan**
# Description

I will be making a new issue once this PR clears to actually train the neural net and get the node fully functional. However since this could be a good chunk of time in the future, I've decided pushing through the changes now makes sense to avoid having the branch go stale. Most of the paths related to the node are placeholders, since we do not have any images to actually train the model on.

This PR does the following:
- Adds multiclass segmentation node
- Adds a launch file for the node
- Adds the ability to train the model in several different ways

Fixes #48 

# Testing steps (If relevant)
## Test Case 1
1. `colcon build && colcon test && colcon test-result --verbose`
2. `. install/setup.bash`
3. `ros2 launch urc_perception multiclass_segmentation.launch.py`

Expectation: Node starts up, but fails due to `No such file or directory: 'Allan/Please/Add/Details'`

# Self Checklist
- [x] I have formatted my code using `ament_uncrustify --reformat`
- [x] I have tested that the new behavior works

### CODE REVIEW

**Comment by a-stickan** [comment]
Going to improve the structure of the PR by splitting it in two
